frontend/public/components/cluster-settings/openid-idp-form.tsx

```diff
@@ -0,0 +1,272 @@
+            required />
+        </div>
+        <div className="co-form-section__separator"></div>
+        <label className="control-label">Claims</label>
+        <p className="co-help-text">The first non-empty claim is used. At least one claim is required.</p>
+        <div className="form-group">
+          <label className="control-label" htmlFor="preferredUserName">Preferred username</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={preferredUserName}
+            aria-describedby="idp-claimusername-help"
+            id="preferredUserName" />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="claimName">Name</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={claimName}
+            aria-describedby="idp-claimname-help"
+            id="claimName" />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="claimEmail">Email</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={claimEmail}
+            aria-describedby="idp-claimemail-help"
+            id="claimEmail" />
+        </div>
+        <div className="co-form-section__separator"></div>
+        <label className="control-label">More options</label>
+        <div className="form-group">
+          <DroppableFileInput
+            onChange={this.caFileChanged}
+            inputFileData={caFileContent}
+            id="caFileContent"
+            label="CA"
+            hideContents />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="extraScopes">Extra scopes</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={extraScopes}
+            aria-describedby="idp-claimname-help"
+            id="extraScopes" />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="extraAuthorizeParameters">Extra authorize parameters</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={extraAuthorizeParameters}
+            aria-describedby="idp-claimemail-help"
+            id="extraAuthorizeParameters" />
+        </div>
+        <ButtonBar errorMessage={this.state.errorMessage} inProgress={this.state.inProgress}>
+          <button type="submit" className="btn btn-primary">Add</button>
+          <button type="button" className="btn btn-default" onClick={history.goBack}>Cancel</button>
+        </ButtonBar>
+      </form>
+    </div>;
+  }
+}
+
+type AddOpenIDPPageState = {
+  name: string;
+  mappingMethod: string;
```

Comment:
This could be a string literal

```suggestion
  mappingMethod: 'claim' | 'lookup' | 'add';
```

We might want to put some IDP types into frontend/public/module/k8s/index.ts (like mapping method) to share across the forms.